### PR TITLE
Stone workbench

### PR DESCRIPTION
**Pull request by MikeStoneColumbia**
Completed Week 2 milestones

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -15,18 +15,79 @@
...
   }
+
+
+  private void toJson(String message){
+
+      if(messages.isEmpty())
+        messages.add("{ \"msg\": \""+message+"\"}");
+    
+      else
+        messages.add("{ \"msg\": \""+message+"\"}");
```
Comment (SimplyFaisal) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:81]: Both of branches of this conditional look identical. Would it be possible to just do `messages.add(....)`?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -15,18 +15,79 @@
...
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    List<String> messages;
```
Comment (SimplyFaisal) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: Would it be beneficial to initialize this when it is declared? I suspect there could be a NullPointException lurking in the code. For example if `toJson()` is called before this field is initialized  on line 52.
```suggestion
    List<String> messages = new ArrayList<>();
```

Comment (SimplyFaisal) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: I thought about this some more and I think this variable should be local to the doGet method. The servlet only gets instantiated once so messages would keep accumulating here until the server is shut down.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -13,11 +13,18 @@
...
-  fetch('/data').then(response => response.text()).then(data => {
+
+  fetch('/data')
+  .then(response => response.json())
+  .then(data => {
     
-    document.getElementById("message").innerHTML += data
+    data.forEach(dataum => {
+      
+      document.getElementById("userMessages").innerHTML += `<p><strong> A user commented:</strong> ${dataum.msg} </p>`
```
Comment (SimplyFaisal) [inline portfolio/src/main/webapp/script.js:25]: Nice use of template strings!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (MikeStoneColumbia) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:81]: Yeah Sounds good I just forgot to remove that, thanks for pointing that out.

**Pull request merged**